tools/rpkg/tests/testthat/test_register_arrow.R

```diff
@@ -446,3 +480,33 @@ test_that("duckdb can read arrow timestamps", {
   }
 })
 
+test_that("duckdb can read arrow timestamptz", {
+  con <- DBI::dbConnect(duckdb::duckdb(), timezone_out = "UTC")
+  on.exit(dbDisconnect(con, shutdown = TRUE))
+
+  timestamp <- as.POSIXct("2022-01-30 11:59:29")
+
+  for (unit in c("s", "ms", "us", "ns")) {
+    tbl <- arrow::arrow_table(t = arrow::Array$create(timestamp, type = arrow::timestamp(unit, "UTC")))
+    duckdb::duckdb_register_arrow(con, "timestamps", tbl)
+
+    if (unit == "ns") {
+      # warning when precision loss
+      expect_warning({ res <- dbGetQuery(con, "SELECT t FROM timestamps") })
+    } else {
+      expect_warning({ res <- dbGetQuery(con, "SELECT t FROM timestamps") }, regexp = NA)
+    }
+    expect_equal(res[[1]], as.POSIXct(as.character(timestamp), tz = "UTC"))
+
+    res <- dbGetQuery(con, "SELECT year(t), month(t), day(t), hour(t), minute(t), second(t) FROM timestamps")
```

Comment:
To fix the failing test here, you probably need to add:
```sql
require icu;
set TimeZone='UTC';
```

Reply:
I think we need to add the UTC extension to the R package